crates/stellar-xdr-wasm/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+mod utils;
+
+use std::{
+    cmp,
+    io::{self, Read},
+    str::FromStr,
+};
+
+use stellar_xdr::{
+    curr::{self, TypeVariant, WriteXdr},
+    schema::Schema,
+};
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn type_variants() -> String {
+    format!(
+        "[{}]",
+        TypeVariant::VARIANTS_STR
+            .iter()
+            .map(|s| format!("{s:?}"))
+            .collect::<Vec<String>>()
+            .join(", ")
+    )
+}
+
+// #[cfg(feature = "schema")]
+#[wasm_bindgen]
+pub fn schema(type_variant: &str) -> Result<String, JsValue> {
+    let Schema(json_schema): Schema = type_variant
+        .parse()
+        .map_err(|_| JsValue::from_str("unknown type"))?;
+    serde_json_wasm::to_string(&json_schema).map_err(|e| JsValue::from_str(&format!("{e:?}")))
+}
+
+#[wasm_bindgen]
+pub fn from_xdr(xdr_base64: String, variant: Option<String>) -> String {
+    let mut f = curr::Limited::new(ResetRead::new(xdr_base64.as_bytes()), curr::Limits::none());
+    for variant in variant
+        .map(|v| Some(vec![v.parse::<crate::curr::TypeVariant>().ok()?]))
+        .unwrap_or_else(|| Some(TypeVariant::VARIANTS.to_vec()))
+        .unwrap()
```

Comment:
I think the from_xdr function, by providing an auto-search for a value, is providing higher level functionality at a lower level than we need to. If we instead expose a `pub fn types() -> Vec<TypeVariant>`, then any caller will be able to iterate over the variants and try them out. Some advantages: keep the logic in the lib simpler, no need to maintain and document string values that have a special form `{"error":...}` that could actually collide with future valid XDR json.